app/services/GetBusinessDetailsService.scala

```diff
@@ -34,6 +35,9 @@ class GetBusinessDetailsService @Inject()(val getBusinessDetailsConnector: GetBu
       case success: IncomeSourceDetailsModel =>
         logger.debug(s"[getBusinessDetailsService][getBusinessDetails] - Retrieved Get Business Details:\n\n$success")
         success
+      case notFound: IncomeSourceDetailsNotFound =>
```

Comment:
* it seems you are overwriting result with error in line 40;
* you should really propagate this change up to controller level, no need to override this ;

Reply:
Done